sharktank/sharktank/layers/paged_llama_attention_block.py

```diff
@@ -108,9 +137,95 @@ def __init__(
                 RMSNormLayer(theta("attn_output_norm"), epsilon=rms_epsilon),
             )
 
+    def pre_process_attention(
+        self,
+        x: torch.Tensor,
+        start_index: int,
+        embedding: RotaryEmbeddingLayer,
+        embedding_batch_mask: torch.Tensor,
+    ):
+
+        if self.attn_type == "mla":
+            if self.wq is not None:
+                q = self.wq(x).unflatten(2, (self.head_count, -1))
+            else:
+                q = self.wq_b(self.q_norm(self.wq_a(x)))
+                if not isinstance(q, ReplicatedTensor):
```

Comment:
```suggestion
                if isinstance(q, SplitPrimitiveTensor):
```

Allows unsharded models to be exported.